src/main/java/Parser.java

```diff
@@ -0,0 +1,123 @@
+import java.util.ArrayList;
+
+public class Parser {
+
+
+    /**
+     * Method that parses user input into instructions to be followed.
+     * @param input the user input
+     * @param ui current instance of ui so that ui can interact with user
+     * @param tasks the tasks available in current Duke program
+     */
+    public void parse(String input, Ui ui, TaskList tasks) {
+        if (input.equals("list")) {
+            ui.printList();
+        }
+
+        String command = "";
+
+        try {
+            command = getCommand(input);
+        } catch(DukeException d) {
+            System.out.println(d.getMessage());
+        }
+
+        switch (command) {
```

Comment:
Best to have the contents of each case in separate functions, it would look more organized.